src/SOFe/RwLock/RwLock.php

```diff
@@ -4,112 +4,63 @@
 
 namespace SOFe\RwLock;
 
-use function assert;
-use function count;
-
 use Closure;
 use Generator;
-use Throwable;
 use SOFe\AwaitGenerator\Await;
 
 /**
- * @phpstan-import-type AsyncRunnable from RwLock
- * @phpstan-import-type Runnable from RwLock
+ * A Read-Write lock to be used in DB initialization
```

Comment:
db initialization is a specific use case, not a good idea to include it in the first line of docs
```suggestion
 * A lock that allows shared or exclusive acquisition.
```